reference-architecture/azure-ansible/bastion.sh

```diff
@@ -0,0 +1,337 @@
+#!/bin/bash
+
+export RESOURCEGROUP=$1
+export AUSERNAME=$2
+export PASSWORD=$3
+export HOSTNAME=$4
+export NODECOUNT=$5
+export ROUTEREXTIP=$6
+export RHNUSERNAME=$7
+export RHNPASSWORD=$8
+export RHNPOOLID=$9
+export SSHPRIVATEDATA=${10}
+export SSHPUBLICDATA=${11}
+export SSHPUBLICDATA2=${12}
+export SSHPUBLICDATA3=${13}
+
+ps -ef | grep bastion.sh > cmdline.out
+
+mkdir -p /home/$AUSERNAME/.ssh
+echo $SSHPUBLICDATA $SSHPUBLICDATA2 $SSHPUBLICDATA3 >  /home/$AUSERNAME/.ssh/id_rsa.pub
+echo $SSHPRIVATEDATA | base64 --d > /home/$AUSERNAME/.ssh/id_rsa
+chown $AUSERNAME /home/$AUSERNAME/.ssh/id_rsa.pub
+chmod 600 /home/$AUSERNAME/.ssh/id_rsa.pub
+chown $AUSERNAME /home/$AUSERNAME/.ssh/id_rsa
+chmod 600 /home/$AUSERNAME/.ssh/id_rsa
+
+mkdir -p /root/.ssh
+echo $SSHPRIVATEDATA | base64 --d > /root/.ssh/id_rsa
+echo $SSHPUBLICDATA $SSHPUBLICDATA2 $SSHPUBLICDATA3   >  /root/.ssh/id_rsa.pub
+chown root /root/.ssh/id_rsa.pub
+chmod 600 /root/.ssh/id_rsa.pub
+chown root /root/.ssh/id_rsa
+chmod 600 /root/.ssh/id_rsa
+
+sleep 30
+cat <<EOF > /root/setup_ssmtp.sh
```

Comment:
Have you tested failing the mail step? The reason why I ask is that it would be a bummer if the installation hung just due to an email. Also, is the email portion required or optional? Could it be configured to be optional if it is currently required?